### ISSUE TITLE
Example for 'MemberName'

### ISSUE THREAD

**Issue by sharwell** [EWSoftware/SHFB#19]
I'm interested in including a link on the documentation for each code element that allows a user to easily request sample code. The simplest case might be a link of the following form:

[Request an example for this page][1]

It would be interesting but not essential to extend this functionality to include information about the commit hash from which the documentation was created, the source file for the member, etc.

:bulb: Since I don't have any specific ideas for an implementation, I'm open to suggestions regarding different ways I might proceed in exposing this functionality to users as easily (for the users) as possible.

[1]: https://github.com/EWSoftware/SHFB/issues/new?title=Example%20for%20%27MemberName%27&body=Please%20include%20an%20example%20for%20`MemberName`.%0D%0D*TODO%20(optional):%20Describe%20a%20specific%20scenario%20you%20would%20like%20to%20see%20addressed*%0D%0D---%0D%0DMember:%20`FullyQualifiedMemberName`%0DPage:%20http://url.to/docs/html/page.htm

**Comment by EWSoftware** [comment]
The easiest way to accomplish this is probably just to add the necessary XSL to the transformations and define transformation arguments on the presentation style that can pass in the variable information such as a project/e-mail URL, an optional commit hash, and one to enable or disable the feature.  A resource item could be added to contain the overall layout and makes the text parts localizable similar to the feedback e-mail link.  See the logo parameters for an example of the transformation arguments.

I doubt you can get source file information for the member without a fair amount of effort since that isn't part of the standard assembly metadata.  You'd probably have to create a plug-in that parses the PDB information if available to add that info to the reflection data.  I'm pretty sure I've seen code that can do that but it's not something I've tried before.

**Comment by sharwell** [comment]
Thanks. :smile: My first goal is to delete my current SHFB repository and then fork this one so I have the ability to send pull requests. For new features this looks like the highest-value win for readers of our documentation.

**Comment by EWSoftware** [comment]
It turns out that MRefBuilder already had the necessary interfaces implemented to read PDB files, it just wasn't putting them to any use.  As such, with a few additions, I was able to get it to generate source file and line number information in the reflection data.  There are a few exceptions since .NET PDB files only contain sequence points for members that contain actual code.  By falling back to a simple search for the type's source code file in cases where it doesn't have that info but the code is structured in folders and files that mimic the namespaces and types, it can usually find a match and works pretty well.

In addition to the *View Source* link I also added a *Request Example* link which should cover the request above.  I've published updates to the SHFB help content so that it can be seen [in action](http://ewsoftware.github.io/XMLCommentsGuide/html/f796665e-e843-4815-cf04-ec60fe45daae.htm).
